### DISCUSSION TITLE
Low Quality when compared with other map

### DISCUSSION THREAD

**Discussion by chmieleski** [visgl/deck.gl#9006] [Q&A]
Hi, I'm having an issue where a map is being rendered with a lower resolution.
It appears that the canvas is not being rendered correctly.

You can check this print:

![image](https://github.com/visgl/deck.gl/assets/11591805/c28e0676-ad5d-469d-b6a9-62b54c17d338)

However not only the TileLayer is in low quality, but any other layer, see example:

![image](https://github.com/visgl/deck.gl/assets/11591805/472f47e1-583c-4263-898f-4a18262814a8)

**Comment by Pessimistress** [comment]
What is this "other map" that you are comparing to?

**Reply by chmieleski** [reply]
I'm using Leaflet, but both of them is using tile API from OMS.

**Reply by chmieleski** [reply]
After checking the docs, I can see that even in examples the map is blurry.

![image](https://github.com/visgl/deck.gl/assets/11591805/adb76b7d-0b11-45f6-9910-c063f1d6a748)

**Reply by Pessimistress** [reply]
I can't tell what other layers you are using from the screenshot.

If you are using the TileLayer with raster tiles you can adjust [tileSize](https://deck.gl/docs/api-reference/geo-layers/tile-layer#tilesize).

For adjusting resolution across the board you may try [useDevicePixels](http://deck.gl/docs/api-reference/core/deck#usedevicepixels)

**Reply by chmieleski** [reply]
The canvas has kind of a weird zoom

![image](https://github.com/visgl/deck.gl/assets/11591805/070d974b-6300-45ef-a535-7ad0e487f55b)

here is the same data map from OMS official site

![image](https://github.com/visgl/deck.gl/assets/11591805/9688404c-3075-4c59-86e0-44069ce8f016)

**Reply by chmieleski** [reply]
> I can't tell what other layers you are using from the screenshot.
> 
> If you are using the TileLayer with raster tiles you can adjust [tileSize](https://deck.gl/docs/api-reference/geo-layers/tile-layer#tilesize).
> 
> For adjusting resolution across the board you may try [useDevicePixels](http://deck.gl/docs/api-reference/core/deck#usedevicepixels)

Thanks, but still no improvement.

**Reply by Pessimistress** [reply]
`tileSize` may not behave the way you want it to, but if you are not seeing any difference then most definitely you are not setting it correctly.

**Reply by chmieleski** [reply]
Thanks, but still not very content with the quality of the canvas. I'm going to publish a page to compare. It's clearly when you put side to side

**Reply by am2222** [reply]
@koka0012 any updates on this? 
I noticed the same too when compared with leaflet or mapbox

**Comment by Pessimistress** [comment]
Ok I took some time to look into this. Try add the following `textureParameters` to your `BitmapLayer`:

```ts
        new BitmapLayer(otherProps, {
          image: data,
          bounds: [west, south, east, north],
          textureParameters: {
            magFilter: 'nearest',
          }
        }),
```

[CodePen](https://codepen.io/Pessimistress/pen/MWMWWqa)

**Reply by am2222** [reply]
@Pessimistress thanks for the example.  Something I noticed if I use a mapbox layer to dexkgl with vector tile text it has a bit of bluriness in it too. Is it any way to avoid it?
